### PR TITLE
Refactored enum use case to PHP 8.1 native enum

### PR DESCRIPTION
**Pull request by DHouweling**

### CODE REVIEW

File: src/HasStateTransitions.php [hunk 1]
```diff
@@ -47,7 +47,7 @@ public function transitionStateTo(TransitionState $to, string $field = null): se
     /**
      * @return array<\LenderSpender\StateTransitionWorkflow\TransitionState>
      */
-    public function getAvailableStateTransitions(?string $field = null): array
+    public function getAvailableStateTransitions(string $field = null): array
```
Comment (TijmenWierenga) [inline src/HasStateTransitions.php:50]: I've seen you remove the nullable type on multiple occasions with a `null` default value. Do you have a specific reason to do so?

File: src/TransitionState.php [hunk 2]
```diff
@@ -6,5 +6,5 @@
 
 interface TransitionState
 {
-    public function __toString();
+    public function getValue(): string;
```
Comment (TijmenWierenga) [inline src/TransitionState.php:9]: You can also extend from the `Stringable` interface in order to restore existing behaviour.

I don't specifically like `getValue` as the method name here. It's basically returning the name of the transition. Maybe `getKey` or `getName` are better alternatives?

File: src/HasStateTransitions.php [hunk 3]
```diff
@@ -47,7 +47,7 @@ public function transitionStateTo(TransitionState $to, string $field = null): se
     /**
      * @return array<\LenderSpender\StateTransitionWorkflow\TransitionState>
```
Comment (TijmenWierenga) [inline src/HasStateTransitions.php:48]: Fixing this generic by specifying the key as a first argument will possibly resolve some ignored PHPStan errors.

File: tests/Stubs/TransitionableModel.php [hunk 4]
```diff
@@ -17,8 +17,8 @@ class TransitionableModel extends Model
     public bool $isTransitionedByCustomWorkflow = false;
     public int $timesTransitionedByQueuedWorkflow = 0;
 
-    /** @var array<string, class-string<\LenderSpender\LaravelEnums\Enum>> */
-    protected $enums = [
+    /** @var string[] */
```
Comment (TijmenWierenga) [inline tests/Stubs/TransitionableModel.php:20]: This is more accurate:

```suggestion
    /** @var array<string, class-string<\BackedEnum>> */
```

**Review by TijmenWierenga — CHANGES_REQUESTED** [review]

File: src/TransitionState.php (hunk 2 above)
Comment (DanielGSoftware) [inline src/TransitionState.php:9]: Now, this is out of scope, just wanted to put my thoughts out there of something that I noticed:

In our own domain we can always assume that only an enum will implement this `TransitionState` interface. I get that this is done so other's can implement their own way and it's not bound to an enum, but is it worth it considering that we have 0 other's using this package? 😀 

Nothing to change for now, but I was just wondering personally 😄

**Review by DanielGSoftware — APPROVED** [review]

File: src/HasStateTransitions.php (hunk 1 above)
Comment (DHouweling) [reply src/HasStateTransitions.php:50]: Didn't know I did this, apparently my phpcs does this.

**Review by TijmenWierenga — APPROVED** [review]

**Review by huubvdw — APPROVED** [review]

**Pull request merged**